api/go.mod

```diff
@@ -1,14 +1,16 @@
 module github.com/red-hat-storage/ocs-operator/api/v4
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.4
```

Comment:
why do we need a seperate toolchain version

Reply:
It was coming up automatically after the dependency upgrades, now removed after seeing this in go documentation here https://go.dev/doc/toolchain.

> If the toolchain line is omitted, the module or workspace is considered to have an implicit toolchain goV line, where V is the Go version from the go line.
> 
> For example, a go.mod that says go 1.21.0 with no toolchain line is interpreted as if it had a toolchain go1.21.0 line.